metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/sampling/UniformSampler.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.statistics.sampling;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class UniformSampler implements Sampler {
+  private List<Object> reservoir;
+  private int seen = 0;
+  private int size;
+  private Random rng = new Random(0);
+
+  public UniformSampler() {
+    this(DEFAULT_SIZE);
+  }
+
+  public UniformSampler(int size) {
+    this.size = size;
+    reservoir = new ArrayList<>(size);
+  }
+
+  @Override
+  public Iterable<Object> get() {
+    return reservoir;
+  }
+
+  /**
+   * Add an object to the reservoir
+   * @param o
+   */
+  public void add(Object o) {
+    if(o == null) {
+      return;
+    }
+    if (reservoir.size() < size) {
+      reservoir.add(o);
+    } else {
+      int rIndex = rng.nextInt(seen + 1);
```

Comment:
Just so I'm clear, up to the reservoir size, we add to the reservoir.  When we're past the reservoir, we do a random replacement as per https://en.wikipedia.org/wiki/Reservoir_sampling

Reply:
you are 100% right, that's what I get for skim reading.

Reply:
Shouldn't we reference Reservoir Sampling in the documentation?  Then the use of Universal and other terms would be more in context.

Reply:
This makes me think that we need "namespace" scoped documentation

Reply:
I modified the docs to have a link to the Reservoir sampling wikipedia article and made things a bit more clear regarding Uniform.